### PR TITLE
POC - proof of concept for uptime statistics

### PR DESCRIPTION
**Pull request by matt-raffel-kiva**
proof of concept for uptime statistics

please see the uptime statistics feature brief for details how this works.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -5,10 +5,7 @@ import { AppService } from './app/app.service';
 
 const bootstrap = async () => {
     const port = process.env.PORT;
-    // Need to disable body parser for http-proxy to work for POSTs: https://github.com/nestjs/nest/issues/405
-    const app = await NestFactory.create(AppModule, {
-        bodyParser: false,
-    });
+    const app = await NestFactory.create(AppModule);
```
Comment (natesulat) [inline src/main.ts:8]: No concerns with this, but for my own education can you just explain the change?

File: src/stats/dtos/service.report.dto.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+
+export class ServiceReportDto {
```
Comment (natesulat) [inline src/stats/dtos/service.report.dto.ts:7]: Nit: Can we delete the first two lines?

File: src/stats/stats.service.ts [hunk 3]
```diff
@@ -1,9 +1,65 @@
...
+
+    constructor(httpService: HttpService) {
+        this.http = new ProtocolHttpService(httpService);
+    }
+
+    public async generateReport(): Promise<SummaryReportDto> {
+
+        Logger.info(`System statistics report generated`);
+
+        let report: SummaryReportDto = new SummaryReportDto();
```
Comment (natesulat) [inline src/stats/stats.service.ts]: Nit: This could be a `const`.

**Review by natesulat — APPROVED** [review]
Looks good. Just so I know how things are supposed to work, how often are we generating a report? Wondering if it will be often enough that we can pinpoint what time a service goes down.

File: src/main.ts (hunk 1 above)
Comment (matt-raffel-kiva) [reply src/main.ts:8]: Ah.  Making it consistent with https://github.com/kiva/aries-guardianship-agency/blob/main/src/main.ts

File: src/stats/stats.service.ts (hunk 3 above)
Comment (matt-raffel-kiva) [reply src/stats/stats.service.ts]: done

File: src/stats/dtos/service.report.dto.ts (hunk 2 above)
Comment (matt-raffel-kiva) [reply src/stats/dtos/service.report.dto.ts:7]: replaced with comment.

File: src/main.ts (hunk 1 above)
Comment (natesulat) [reply src/main.ts:8]: Right on. Thanks!

**Review by mac-arrap — APPROVED** [review]

**Review by natesulat — APPROVED** [review]
We should probably write tests for this at some point

**Pull request merged**